### PR TITLE
Add unique pathname to each event/method and some tweaks

### PR DESCRIPTION
**Pull request by peterbroadhurst**
As discussed with @nguyer I needed to add unique `pathname`s to each FFI method/event so that I can generate swagger.
This also would be required for other paths that need unique names.

As I went few I ended up proposing a few more changes:
- Added `description` fields to FFI/FFIMethod/FFIEvent
  - These are just text, and would go into the generate Swagger
  - For Eth ABI -> FFI these would come from the Devdocs
- The broadcast wasn't inserting events, only methods. So fixed that
- Checked for empty names on event/methods
- Changed index to be on `pathname` (not `name`)
  - Changed all lookups by name to this
- Observed `ContractSubscription` code was inserting with `nil` contract ID on `FFIEvent`
  - This would not work, as you could fail to subscribe if you created multiple subs on the same event name
  - Moved to in-line copy of the event definition on all subscriptions
  - Allowed either in-line or reference an event by ID within the same namespace
- Tweaked SQL layer for consistency to allow query of interfaceid, and return all fields (inc interface ID and namespace)
  - Seemed problematic to have special rules at the DB implementation layer to redact some fields
  - If we have a query layer that wants to redact these when they are nested in-line in a parent object, that makes sense to do consistently there. But other code that queries the FFIMethod/FFIEvent directly should get all fields
  - When we get to NoSQL that would add complexity
  - _I left a piece I couldn't understand, which was why the update path only updated the `params` (not the other fields)_

> I observed we only have SQLite migrations at the moment - important to remember to address this before raising a PR from the feature branch, up to the HL main.

### CODE REVIEW

**Comment by nguyer** [comment]
Andrew made some changes to the `ContractSubscription` in https://github.com/kaleido-io/firefly/pull/45/files which allows for duplicates of a `nil` name in the DB. I think we have to allow not having a `ContractID` on a `ContractSubscription`, because it's possible to listen to contract events without uploading an FFI. Or am I misunderstanding something from your change?

**Comment by peterbroadhurst** [comment]
The change I made was to distinguish between a table stored FFIEvent, which is a definition of an event on an interface (which has uniqueness within an FFI, and we’ve discussed could change over time), and the details of an event in a subscription, which with this proposal is simply stored with the subscription.

File: internal/oapispec/ffi2swagger.go [hunk 1]
```diff
@@ -35,9 +35,35 @@ func NewFFISwaggerGen() FFISwaggerGen {
...
+	// 		return nil, err
+	// 	}
+	// }
+
+	// for _, event := range ffi.Events {
+	// 	if routes, err = og.addEvent(ctx, routes, event); err != nil {
+	// 		return nil, err
+	// 	}
+	// }
+
```
Comment (nguyer) [inline internal/oapispec/ffi2swagger.go:53]: Is this going to be uncommented at some point, or should it be removed?

File: internal/oapispec/ffi2swagger.go [hunk 2]
```diff
@@ -35,9 +35,35 @@ func NewFFISwaggerGen() FFISwaggerGen {
...
+// 		Name: fmt.Sprintf("method_%s_invoke", method.Name),
+// 	})
+
+// 	return routes, nil
+// }
+
+// func (og *ffiSwaggerGen) addEvent(ctx context.Context, routes []*Route, event *fftypes.FFIEvent) ([]*Route, error) {
+
+// 	return routes, nil
+// }
```
Comment (nguyer) [inline internal/oapispec/ffi2swagger.go:69]: Is this going to be uncommented at some point, or should it be removed?

**Review by nguyer — COMMENTED** [review]
These changes make sense to be but it would probably be good to have @awrichar review them as well.

File: internal/oapispec/ffi2swagger.go (hunk 2 above)
Comment (peterbroadhurst) [reply internal/oapispec/ffi2swagger.go:69]: Either way - this was the key thing I wanted to work on (was hoping to have it done by today sorry), but needed the unique pathnames, which took me down the rabbit hole with the other changes.

Comment (peterbroadhurst) [reply internal/oapispec/ffi2swagger.go:69]: I submitted the PR early into the working branch, as I felt I needed to discuss with @awrichar the approach on the events.

Comment (nguyer) [reply internal/oapispec/ffi2swagger.go:69]: Ah got it. We can leave it for now, since this is just a PR into the feature branch

**Comment by peterbroadhurst** [comment]
I note the `subscription` branch from #45 (thanks for pointing me at that) does not change the database to allow duplicates, it also doesn't address deletion.
So if we have a different proposal for storing the events by reference (rather than in-line), I'm happy to work that through:
https://github.com/kaleido-io/firefly/blob/3f185db0b5842f3efe5acb2024a1ff68697e3d64/db/migrations/sqlite/000050_create_ffi_events_table.up.sql#L10

File: db/migrations/sqlite/000048_create_ffi_table.up.sql [hunk 3]
```diff
@@ -3,7 +3,8 @@ CREATE TABLE ffi (
   id                UUID            NOT NULL,
   namespace         VARCHAR(64)     NOT NULL,
   name              VARCHAR(1024)   NOT NULL,
-  version           VARCHAR(64)     NOT NULL
+  version           VARCHAR(64)     NOT NULL,
+  description       VARCHAR(65536)  NOT NULL
```
Comment (awrichar) [inline db/migrations/sqlite/000048_create_ffi_table.up.sql:7]: Should we just use a `TEXT` column here?

**Comment by awrichar** [comment]
Seems like a good route to me 👍

**Comment by peterbroadhurst** [comment]
Thanks @awrichar - I've merged with your changes now, if you'd like to pull this into the main `onchain-logic` branch

**Pull request merged**